src/test/java/LoginTest.java

```diff
@@ -0,0 +1,114 @@
+import org.testng.Assert;
+import org.testng.annotations.Test;
+import page.HomePage;
+//import page.LoginPage;
+import pageFactory.LoginPage;
+
+public class LoginTest extends BaseTest {
+
+
+@Test
+    public void loginEmptyEmailPassword()  {
+
+        LoginPage loginPage = new LoginPage(driver);
+        HomePage homePage = new HomePage(driver);
+
+        loginPage.provideEmail(" ").provideEmail("jKV0uSX6z1dv").clickSubmit();
+
+        String expectedURL= "https://qa.koel.app/";
+        Assert.assertEquals(driver.getCurrentUrl(), expectedURL);
+
+        driver.quit();
+        /*
+        String expectedURL= "https://qa.koel.app/";
+        //navigateToPage();
+        enterEmail("");
+        enterPassword("jKV0uSX6z1dv");
+
+ */
+    }
+@Test
+
+    public void loginValidEmailPassword()  {
+
+   LoginPage loginPage = new LoginPage(driver);
+   HomePage homePage = new HomePage(driver);
+
+   // login has step in it with a method
+
+loginPage.provideEmail("christina.taylor@testpro.io").providePassword("jKV0uSX6z1dv").clickSubmit();
+
+ Assert.assertTrue(homePage.getUserAvatarIcon().isDisplayed());
+
+
+
+
+
+
+
+
+
+
```

Comment:
**Please delete the empty line!** An Empty Line is a "Code Smell"
The short answer is, _They’re more an annoyance than a problem.They won’t actually break anything—which is why your code continues to run—but they make it awkward to read and change your code, especially if you’re dealing with multiple line breaks per line of real code. They also add just a tiny bit of size to your files, which is nice to reduce._

Reply:
so just to be clear remove al the notes example "//code"

Reply:
ok i did not see the bottom message ok let me fix